physics/cam_diagnostics.F90

```diff
@@ -365,6 +379,364 @@ subroutine diag_init_dry(pbuf2d)
+      call addfld ('DGT_SS  '  ,horiz_only, 'A','unitless','sea-salt aerosol optical depth 550nm gt05')     
+      call addfld ('DGT_SO4 '  ,horiz_only, 'A','unitless','SO4 aerosol optical depth 550nm gt05')     
+      call addfld ('DGT_POM '  ,horiz_only, 'A','unitless','OC aerosol optical depth 550nm gt05')     
+      call addfld ('DGT_BC  '  ,horiz_only, 'A','unitless','BC aerosol optical depth 550nm gt05')     
+      call addfld ('AIRMASS '  ,horiz_only, 'A','kg/m2   ','Vertically integrated airmass')     !akc6
+
+      call addfld ('NNAT_0  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 0 number concentration')     
+      call addfld ('NNAT_1  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 1 number concentration')     
+      call addfld ('NNAT_2  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 2 number concentration')     
+      call addfld ('NNAT_4  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 4 number concentration')     
+      call addfld ('NNAT_5  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 5 number concentration')     
+      call addfld ('NNAT_6  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 6 number concentration')     
+      call addfld ('NNAT_7  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 7 number concentration')     
+      call addfld ('NNAT_8  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 8 number concentration')     
+      call addfld ('NNAT_9  ',(/'lev'/),'A','1/cm3   ','Aerosol mode 9 number concentration')     
+      call addfld ('NNAT_10 ',(/'lev'/),'A','1/cm3   ','Aerosol mode 10 number concentration')     
+      call addfld ('NNAT_12 ',(/'lev'/),'A','1/cm3   ','Aerosol mode 12 number concentration')     
+      call addfld ('NNAT_14 ',(/'lev'/),'A','1/cm3   ','Aerosol mode 14 number concentration')     
+      call addfld ('AIRMASSL',(/'lev'/),'A','kg/m2   ','Layer airmass')     
+      call addfld ('BETOTVIS',(/'lev'/),'A','1/km'    ,'Aerosol 3d extinction at 0.442-0.625') ! CAM4-Oslo: 0.35-0.64um
+      call addfld ('BATOTVIS',(/'lev'/),'A','1/km'    ,'Aerosol 3d absorption at 0.442-0.625') ! CAM4-Oslo: 0.35-0.64um
+      call addfld ('BATSW13 ',(/'lev'/),'A','1/km'    ,'Aerosol 3d SW absorption at 3.077-3.846um')
+      call addfld ('BATLW01 ',(/'lev'/),'A','1/km'    ,'Aerosol 3d LW absorption depth at 3.077-3.846um')
+
+      do i=1,nbmodes
+         modeString="  "
+         write(modeString,"(I2)"),i
+         if(i.lt.10) modeString="0"//adjustl(modeString)
+         varName = "Camrel"//trim(modeString)
+         if(i.ne.3) call addfld(varName, (/'lev'/),'A','unitless', 'relative added mass for mode'//modeString)
+      enddo  
+      do i=1,nbmodes
+         modeString="  "
+         write(modeString,"(I2)"),i
+         if(i.lt.10) modeString="0"//adjustl(modeString)
+         varName = "Cxsrel"//trim(modeString)
+         if(i.ne.3) call addfld(varName, horiz_only, 'A', 'unitless', 'relative exessive added mass column for mode'//modeString)
```

Comment:
Put the `varName` code above inside this `if` statement. Why compute it if you are not going to use it?